src/Libraries/Microsoft.Extensions.Http.Resilience/Routing/Internal/RoutingResilienceStrategy.cs

```diff
@@ -31,6 +31,11 @@ public RoutingResilienceStrategy(Func<RequestRoutingStrategy> provider)
             Throw.InvalidOperationException("The HTTP request message was not found in the resilience context.");
         }
 
+        if (_provider == null)
+        {
+            return await callback(context, state).ConfigureAwait(context.ContinueOnCapturedContext);
```

Comment:
Do we need to await?

Reply:
Yes, as we need to dispose the `RequestRoutingStrategy` after callback is executed.